libcnb-cargo/src/main.rs

```diff
@@ -0,0 +1,187 @@
+// https://rust-lang.github.io/rust-clippy/stable/index.html
+#![warn(clippy::pedantic)]
+// This lint is too noisy and enforces a style that reduces readability in many cases.
+#![allow(clippy::module_name_repetitions)]
+
+mod cli;
+use cargo_metadata::MetadataCommand;
+use clap::ArgMatches;
+use libcnb_cargo::cross_compile::cross_compile_help;
+use libcnb_cargo::{
+    assemble_buildpack_directory, build_buildpack_binary, default_buildpack_directory_name,
+    read_buildpack_data, BuildError, BuildpackDataError, CargoProfile,
+};
+use log::error;
+use log::info;
+use size_format::SizeFormatterSI;
+use std::path::PathBuf;
+
+fn main() {
+    setup_logging();
+
+    match cli::setup_cli_parsing().get_matches().subcommand() {
+        ("libcnb", Some(matches)) => match matches.subcommand() {
+            ("package", Some(matches)) => handle_libcnb_package(matches),
+            // This should never be reached since clap will catch unknown subcommands for us
+            _ => unimplemented!("Only the \"package\" subcommand is currently implemented!"),
+        },
+        // This should never be reached since clap will catch unknown subcommands for us
+        _ => unimplemented!("Only the \"libcnb\" subcommand is currently implemented!"),
+    }
+}
+
+#[allow(clippy::too_many_lines)]
+fn handle_libcnb_package(matches: &ArgMatches) {
+    let cargo_profile = if matches.is_present("release") {
+        CargoProfile::Release
+    } else {
+        CargoProfile::Dev
+    };
+
+    let target_triple = match matches.value_of("target") {
+        None => {
+            error!("Could not determine target triple!");
+            std::process::exit(1);
+        }
+        Some(target_triple) => target_triple,
+    };
+
+    let current_dir = match std::env::current_dir() {
+        Ok(current_dir) => current_dir,
+        Err(io_error) => {
+            error!("Could not determine current directory: {}", io_error);
+            std::process::exit(1);
+        }
+    };
+
+    info!("Reading buildpack metadata...");
+    let buildpack_data = match read_buildpack_data(&current_dir) {
+        Ok(buildpack_data) => buildpack_data,
+        Err(error) => {
+            match error {
+                BuildpackDataError::IoError(io_error) => {
+                    error!("Unable to read buildpack metadata: {}", io_error);
+                    error!("Hint: Verify that a readable file named \"buildpack.toml\" exists at the root of your project.");
+                }
+                BuildpackDataError::DeserializationError(deserialization_error) => {
+                    error!(
+                        "Unable to deserialize buildpack metadata: {}",
+                        deserialization_error
+                    );
+                    error!("Hint: Verify that your \"buildpack.toml\" is valid.");
```

Comment:
```suggestion
                    error!("Hint: Verify that your \"buildpack.toml\" is valid TOML.");
```

Reply:
This goes beyond valid TOML. It also verifies if the file conforms to the CNB spec. It will fail is required fields from `buildpack.toml` are missing or of a wrong type for example.